File: PKI-web/app/frontend/src/app/models/EventOffer.ts
import { Event } from "./Event";
import { Photo } from "./Photo";

export class EventOffer {
  eventOfferId?: number;
  name!: string;
  description?: string;
  shortDescription!: string;
  price!: number;
  photoId?: Photo;
  eventList?: Array<Event>;

  constructor(init?: Partial<EventOffer>) {
    Object.assign(this, init);
  }
}

File: app/src/main/java/com/example/pki/model/EventOffer.java
package com.example.pki.model;

import androidx.annotation.Nullable;
import java.util.List;

public class EventOffer {
    @Nullable
    public Integer eventOfferId;
    public String name;
    @Nullable
    public String description;
    public String shortDescription;
    public double price;
    @Nullable
    public Photo photoId;
    @Nullable
    public List<Event> eventList;
}
